nucypher/characters/lawful.py

```diff
@@ -141,6 +141,23 @@ def grant(self, bob, uri, m=None, n=None, expiration=None, deposit=None, handpic
         policy.enact(network_middleware=self.network_middleware)
         return policy  # Now with TreasureMap affixed!
 
+    def revoke(self, policy):
+        """
+        Goes through the RevocationKit in Policy and revokes re-encryption
+        access to Bob from each arrangement.
+        """
+        # Sign the RevocationNotices in the RevocationKit in preparation for
+        # sending to Ursula.
+        policy.revocation_kit.sign(self.stamp)
+
+        failed_revocations = list()
+        for node_id, notice in policy.revocation_kit:
+            ursula = self.known_nodes[node_id]
```

Comment:
Hmm - this may not be robust against `KeyError`, for example, if there are no known nodes.

Reply:
Yeah right. If node isn't found, it's also a failed revocation